src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp

```diff
@@ -116,32 +226,278 @@ bool MKLDNNGatherNode::needPrepareParams() const {
+    const uint64_t idxElPerVec = jitKernel->getIdxElPerVec();
+
+    if (afterAxisSize == 1) { // Elementwise gather.
+        if (specIndicesSize >= idxElPerVec)
+            return; // Is not a short case.
+
+        p.permIdxMask.resize(idxElPerVec);
+        p.srcBeforeAxisDiff.resize(idxElPerVec);
+
+        p.permIdxMask[0] = idxElPerVec - specIndicesSize;
+        for (int i = 1; i < idxElPerVec; i++) {
+            p.permIdxMask[i] = p.permIdxMask[i - 1] + 1;
+            if (p.permIdxMask[i] == idxElPerVec)
+                p.permIdxMask[i] = idxElPerVec - specIndicesSize;
+        }
+
+        const int div = idxElPerVec / specIndicesSize;
+        const int remainder = idxElPerVec % specIndicesSize;
+        for (uint64_t i = 0; i < idxElPerVec; i++) {
+            if (((start + i) % specIndicesSize) < (specIndicesSize - remainder)) {
+                p.srcBeforeAxisDiff[i] = axisDim * div;
+            } else {
+                p.srcBeforeAxisDiff[i] = axisDim * (div + 1);
+            }
+        }
+    } else { // Blocked gather.
+        if (afterAxisSize > idxElPerVec)
+            return; // Is not a short case.
+
+        p.afterAxIdxInBytes.resize(idxElPerVec);
+        p.afterAxPermMask.resize(idxElPerVec);
+        p.beforeAxPermMask.resize(idxElPerVec);
+        p.specIdxDiff.resize(idxElPerVec);
+        p.srcBeforeAxisDiff.resize(idxElPerVec);
+
+        int secondStart = start + idxElPerVec;
+        for (int i = 0; i < idxElPerVec; i++) {
+            p.afterAxIdxInBytes[i] = (start + i) % afterAxisSize;
+            p.specIdxDiff[i] = (((secondStart + i) / afterAxisSize) % specIndicesSize) * idxTypeSize - p.specIdxInBytes[i];
+            if (p.specIdxDiff[i] < 0)
+                p.specIdxDiff[i] += specIndicesSize * idxTypeSize;
+            p.srcBeforeAxisDiff[i] = ((start + i + idxElPerVec) / (specIndicesSize * afterAxisSize)) * axisAndAfterAxisSizeInBytes -
+                    ((start + i) / (specIndicesSize * afterAxisSize)) * axisAndAfterAxisSizeInBytes;
+
+            p.afterAxIdxInBytes[i] *= dataTypeSize;
+            p.afterAxPermMask[i] = idxElPerVec - afterAxisSize + i;
+            for (size_t j = 0lu; j < 6lu; j++) {
+                if (p.afterAxPermMask[i] >= idxElPerVec)
+                    p.afterAxPermMask[i] -= afterAxisSize;
+            }
+        }
+        if (specIndicesSize * afterAxisSize < idxElPerVec) {
+            p.beforeAxPermMask[0] = idxElPerVec - specIndicesSize * afterAxisSize;
+            for (int i = 1; i < idxElPerVec; i++) {
+                p.beforeAxPermMask[i] = p.beforeAxPermMask[i - 1] + 1;
+                if (p.beforeAxPermMask[i] == idxElPerVec)
+                    p.beforeAxPermMask[i] = idxElPerVec - specIndicesSize * afterAxisSize;
+            }
+        }
+
+        p.specIdxAndAfterAxIterB = (start * dataTypeSize) % specIdxAndAfterAxSizeB;
+    }
+}
+
+void MKLDNNGatherNode::execReference() {
```

Comment:
Does ref impl has exactly the same behavior as Jit one? For example I don't see there `reverseIndexing` parameter is used.

Reply:
Looks like it was the one difference. Fixed.